engine/src/main/java/io/zeebe/engine/nwe/BpmnElementProcessor.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.nwe;
+
+import io.zeebe.engine.processor.workflow.deployment.model.element.ExecutableFlowElement;
+
+/**
+ * The business logic of a BPMN element.
+ *
+ * <p>The execution of an element is divided into multiple steps that represents the lifecycle of
+ * the element. Each step defines a set of actions that can be performed in this step. The
+ * transition to the next step must be triggered explicitly in the current step.
```

Comment:
Which part is unclear for you?

> The transition to the next step must be triggered explicitly in the current step.

The transition must be triggered explicitly because the next step may depend on the element or the current instance. For example, a multi-instance activity with an empty input collection is completed immediately.

Reply:
I have the feeling we always are blocked by the multi instance :sweat_smile: Could we handle it differently. So for example transform it dynamically to parallel gw and multiple service tasks ? :grin:

Reply:
> Could we handle it differently. So for example transform it dynamically to parallel gw and multiple service tasks ?

Interesting idea but I see some problems. The multi-instance is like a container that can be canceled via boundary event. The current solution, especially with the [new processor](https://github.com/zeebe-io/zeebe/blob/086af69012c81984169acd1c42a25d3007af9569/engine/src/main/java/io/zeebe/engine/nwe/container/MultiInstanceBodyProcessor.java), looks nice so far :wink:

Reply:
> . The multi-instance is like a container that can be canceled via boundary event